pkg/test/component_redis.go

```diff
@@ -4,14 +4,33 @@ import (
 	"fmt"
 	"github.com/go-redis/redis"
 	"log"
-	"sync"
 )
 
 type redisConfig struct {
-	Host string `mapstructure:"host"`
-	Port int    `mapstructure:"port"`
+	Debug bool   `mapstructure:"debug"`
+	Host  string `mapstructure:"host"`
+	Port  int    `mapstructure:"port"`
 }
 
+var redisConfigs map[string]*redisConfig
+var redisClients simpleCache
+
+func init() {
+	redisClients = simpleCache{}
+	redisConfigs = make(map[string]*redisConfig)
+}
+
+func ProvideRedisClient(name string) *redis.Client {
```

Comment:
i would suggest to put the functions in execution order

Reply:
✅